src/wp-includes/embed.php

```diff
@@ -1243,3 +1243,23 @@ function wp_filter_pre_oembed_result( $result, $url, $args ) {
 
 	return $result;
 }
+
+/**
+ * Adds noindex to the robots meta tag for embeds.
+ *
+ * Typical usage is as a {@see 'wp_robots'} callback:
+ *
+ *     add_filter( 'wp_robots', 'wp_embed_no_robots' );
+ *
+ * @since 5.7.0
+ *
+ * @param array $robots Associative array of robots directives.
+ * @return array Filtered robots directives.
+ */
+function wp_embed_no_robots( array $robots ) {
```

Comment:
Same potential fatal here if $robots not an array?

Reply:
Since the filter should only be used with arrays per documentation, I think it's fair to expect an array here. If a developer misuses the filter not returning an array, that is a mistake that IMO should be caught during development and therefore result in a fatal error.

Reply:
Just thinking that WordPress tries to be more forgiving. 

Consider a plugin that is filtering this which may not show up at activation time. Upon going to the frontend, it could cause a fatal error and that would be a bad experience for end users. 

I'd suggest rather:

```suggestion
function wp_embed_no_robots( $robots ) {
	if ( ! is_array( $robots ) ) {
		_doing_it_wrong( 'add_filter', __( 'A plugin has filtered wp_robots to not return an array as required.' ) );
		$robots = array();
	}
```

Reply:
I agree of course that would be a bad experience for users. I think a `_doing_it_wrong` would be appropriate e.g. if the expected behavior of a filter was changed. However, we're introducing a new filter here, and the "forgivingness" in a case like here tends to introduce/support incorrect usage of the API. If a developer writes code that ignores the documentation, they should expect to be informed of the error in an explicit way so that they can fix it.

A fatal error being thrown when a developer tries to use a non-array on the filter immediately shows, even to inexperienced developers, that don't have `WP_DEBUG` / `WP_DEBUG_DISPLAY` enabled, that this is incorrect usage. Which is important because that's how it gets fixed before it even gets into the hands of users.